visual-dotnet/SauceLabs.Visual/Utils/DiffingOptionsInHelper.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using SauceLabs.Visual.GraphQL;
+using SauceLabs.Visual.Models;
+
+namespace SauceLabs.Visual.Utils
+{
+    public static class DiffingOptionsInHelper
+    {
+        private static readonly Dictionary<DiffingOption, Func<DiffingOptionsIn, bool, DiffingOptionsIn>> Mapping = new Dictionary<DiffingOption, Func<DiffingOptionsIn, bool, DiffingOptionsIn>>()
+        {
+            { DiffingOption.Content, (diffingOptions, b) => {
+                diffingOptions.Content = b;
+                return diffingOptions;
+            } },
+            { DiffingOption.Dimensions, (diffingOptions, b) => {
+                diffingOptions.Dimensions = b;
+                return diffingOptions;
+            } },
+            { DiffingOption.Position, (diffingOptions, b) => {
+                diffingOptions.Position = b;
+                return diffingOptions;
+            } },
+            { DiffingOption.Structure, (diffingOptions, b) => {
+                diffingOptions.Structure = b;
+                return diffingOptions;
+            } },
+            { DiffingOption.Style, (diffingOptions, b) => {
+                diffingOptions.Style = b;
+                return diffingOptions;
+            } },
+            { DiffingOption.Visual, (diffingOptions, b) => {
+                diffingOptions.Visual = b;
+                return diffingOptions;
+            } },
+        };
+
+        internal static DiffingOptionsIn? CreateFromEnableOnlyDisable(IReadOnlyCollection<DiffingOption>? enableOnly, IReadOnlyCollection<DiffingOption>? disable)
+        {
+            if (enableOnly?.Count > 0)
+            {
+                var options = new DiffingOptionsIn(false);
+                foreach (var opt in enableOnly)
+                {
+                    if (Mapping.TryGetValue(opt, out var fn))
```

Comment:
Can't we just simply have an update method on `DiffingOptionsIn`? E.g. `options.set(opt, true)`. That should also remove the `Try...`.

Reply:
I wanted to prevent the GraphQL layer to be aware of that enum that has been build on top since this is only providing customer a convenient way to select what to diff.

I can convert it to a `switch/case` may be a cleaner solution if you prefer this.

Reply:
I assume most users don't care about the underlying backend API. I think we should make the API that the user is actually uses safe and easy to use.